compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala

```diff
@@ -165,7 +165,7 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
       val leftArg = if (isRightAssoc && isInfixType(l)) "(" ~ argText(l) ~ ")" else argText(l)
       val rightArg = if (!isRightAssoc && isInfixType(r)) "(" ~ argText(r) ~ ")" else argText(r)
 
-      leftArg ~ " " ~ simpleNameString(op.classSymbol) ~ " " ~ rightArg
+      leftArg ~ " " ~ simpleNameString(op.typeSymbol) ~ " " ~ rightArg
```

Comment:
FWIW can be dropped, this is already fixed in #4072 (https://github.com/lampepfl/dotty/pull/4072/commits/5e01496d910caaa57a0d950863816d40d8443981 and preserved by later refactorings).

Reply:
On rebasing I reverted the conflicting code from #4072. So "("...")" will only be printed if precedence demands it. Not writting redundant parens is important in particular for long sequences of type operators, as for example when working with HLists.

Reply:
@odersky Did you lose your changes? This PR (as merged) contains no such revert, and #4072 *replaced* extra parentheses with proper use of `atPrec`. Either you lost your changes or the code you reverted was the old one.

If you can find a bug in master, happy to fix it and add tests. 😄